tests/models/test_resource.py

```diff
@@ -195,18 +195,37 @@ def test_mirrorable_media_type(self):
         eq_(False, representation.mirrorable_media_type)
 
     def test_guess_media_type(self):
-        m = Representation.guess_media_type
+        m_file = Representation.guess_media_type
+        m_url = Representation.guess_url_media_type_from_path
+        jpg_file = "file.jpg"
+        zip_file = "file.ZIP"
 
-        eq_(Representation.JPEG_MEDIA_TYPE, m("file.jpg"))
-        eq_(Representation.ZIP_MEDIA_TYPE, m("file.ZIP"))
+        eq_(Representation.JPEG_MEDIA_TYPE, m_file(jpg_file))
+        eq_(Representation.ZIP_MEDIA_TYPE, m_file(zip_file))
```

Comment:
Add a test where the filename is an absolute path to a file on disk; I'm concerned that running a file path through `urlparse` might lose the data we're looking for.

Reply:
Added.